Tp operation stats

TPRobots opened the pull request:
Pull request opened by github-pullrequestcreator.

Commit:
Reimplementation of longest running query and index build collector

Commit:
Switch to seconds, fix naming conventions for prometheus units

csibar reviewed (approved):
### Building mongodb_exporter

```
rucs (tp-operation-stats %) mongodb_exporter $ git show -q
commit 7f8467808dfdd72273e7e7b0ce476682cca13bff (HEAD -> tp-operation-stats, origin/tp-operation-stats)
Author: Audrius Karabanovas <auk@trustpilot.com>
Date:   Wed Jun 23 10:27:29 2021 +0300

    Switch to seconds, fix naming conventions for prometheus units
rucs (tp-operation-stats %) mongodb_exporter $ 
rucs (tp-operation-stats %) mongodb_exporter $ 
rucs (tp-operation-stats %) mongodb_exporter $ make build 
go build -ldflags="-X main.version=7f8467808dfdd72273e7e7b0ce476682cca13bff -X main.buildDate=2021-06-30T12:57:10+0200 -X main.commit=7f8467808dfdd72273e7e7b0ce476682cca13bff -X main.Branch=tp-operation-stats -X main.GoVersion=go1.16.5 -s -w"  -o ./mongodb_exporter
go: downloading github.com/alecthomas/kong v0.2.17
go: downloading github.com/percona/percona-toolkit v0.0.0-20210503082911-610ab8c66a81
go: downloading github.com/prometheus/common v0.25.0
go: downloading go.mongodb.org/mongo-driver v1.5.3
rucs (tp-operation-stats %) mongodb_exporter $ 
rucs (tp-operation-stats %) mongodb_exporter $ 
rucs (tp-operation-stats %) mongodb_exporter $ ./mongodb_exporter --version
mongodb_exporter - MongoDB Prometheus exporter
Version: 7f8467808dfdd72273e7e7b0ce476682cca13bff
Commit: 7f8467808dfdd72273e7e7b0ce476682cca13bff
Build date: 2021-06-30T12:58:10+0200
rucs (tp-operation-stats %) mongodb_exporter $ 
```

### Operations metrics from mongod 3.6.23

```
rucs (tp-operation-stats %) mongodb_exporter $ curl -s http://localhost:9001/metrics | grep -E 'mongodb_operations.+{'
mongodb_operations_index_building{cl_id="5b434d5a698d20d2fc5baac6",cl_role="shardsvr",rs_nm="servicereviews",rs_state="2"} 0
mongodb_operations_index_building_progress_done{cl_id="5b434d5a698d20d2fc5baac6",cl_role="shardsvr",rs_nm="servicereviews",rs_state="2"} 0
mongodb_operations_index_building_progress_total{cl_id="5b434d5a698d20d2fc5baac6",cl_role="shardsvr",rs_nm="servicereviews",rs_state="2"} 0
mongodb_operations_longest_running_query_seconds{cl_id="5b434d5a698d20d2fc5baac6",cl_role="shardsvr",rs_nm="servicereviews",rs_state="2"} 0.000118
rucs (tp-operation-stats %) mongodb_exporter $ 
```

### Operations metrics from mongod 4.4.6

```
rucs (tp-operation-stats %) mongodb_exporter $ curl -s http://localhost:9002/metrics | grep -E 'mongodb_operations.+{'
mongodb_operations_index_building{cl_id="608801d270393e6d933c9df9",cl_role="shardsvr",rs_nm="dataplatformtest",rs_state="2"} 0
mongodb_operations_index_building_progress_done{cl_id="608801d270393e6d933c9df9",cl_role="shardsvr",rs_nm="dataplatformtest",rs_state="2"} 0
mongodb_operations_index_building_progress_total{cl_id="608801d270393e6d933c9df9",cl_role="shardsvr",rs_nm="dataplatformtest",rs_state="2"} 0
mongodb_operations_longest_running_query_seconds{cl_id="608801d270393e6d933c9df9",cl_role="shardsvr",rs_nm="dataplatformtest",rs_state="2"} 6.271209
rucs (tp-operation-stats %) mongodb_exporter $ 
```

csibar committed:
Make the tests happy

percona-csalguero committed:
PMM-6877 MongoS recognition (#299)

Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>
Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/reviewdog/reviewdog from 0.12.0 to 0.13.0 in /tools (#315)

Bumps [github.com/reviewdog/reviewdog](https://github.com/reviewdog/reviewdog) from 0.12.0 to 0.13.0.
- [Release notes](https://github.com/reviewdog/reviewdog/releases)
- [Changelog](https://github.com/reviewdog/reviewdog/blob/master/CHANGELOG.md)
- [Commits](https://github.com/reviewdog/reviewdog/compare/v0.12.0...v0.13.0)

---
updated-dependencies:
- dependency-name: github.com/reviewdog/reviewdog
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/tools from 0.1.4 to 0.1.5 in /tools (#316)

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.1.4 to 0.1.5.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.1.4...v0.1.5)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

relzhong committed:
PMM-8489 fix: topology in mongos (#322)

* fix: topology in mongos

* Changelog 322

percona-csalguero committed:
Fix deps for 0.20.7 (#327)

* Revert "PMM-8489 fix: topology in mongos (#322)"

This reverts commit 4b62f33b4a3e409f8669d0b4bdd15400b3c2feed.

* Revert "Revert "PMM-8489 fix: topology in mongos (#322)""

This reverts commit 07ebd7a5fe203a177f9fa40bd350d3dcd71becd2.

* Fixing deps for release

There was an error on deps after merging 3rd aprty contribution.
https://github.com/percona/mongodb_exporter/pull/322

percona-csalguero committed:
Release 0.20.7 (#328)

dependabot[bot] committed:
Bump github.com/golangci/golangci-lint from 1.41.1 to 1.42.0 in /tools (#333)

Bumps [github.com/golangci/golangci-lint](https://github.com/golangci/golangci-lint) from 1.41.1 to 1.42.0.
- [Release notes](https://github.com/golangci/golangci-lint/releases)
- [Changelog](https://github.com/golangci/golangci-lint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/golangci/golangci-lint/compare/v1.41.1...v1.42.0)

---
updated-dependencies:
- dependency-name: github.com/golangci/golangci-lint
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kanakagrawal committed:
PMM-7424 Add dbStatsCollector (#329)

* https://jira.percona.com/browse/PMM-7424 Add dbStatsCollector

* Run make format

* Add dbstats_ prefix

* Update Changelog

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

percona-csalguero committed:
PMM-8539 Fixed cluster role cluster

Fixed following specs in https://jira.percona.com/browse/PMM-6494.
cl_role: Cluster role - "shardsvr", "configsvr", "mongos" or "".
Set "mongos" if there is a sharding.configDB config value set.
For others whatever the value for sharding.clusterRole is.
It will be empty when a non-sharded replicaset or standalone mongod node.

percona-csalguero committed:
PMM-8539 Added docker helpers

kanakagrawal committed:
Provide old mongod metrics for dbstats (#337)

* Provide old mongod metrics for dbstats

* Fix test case

* Add Changelog

hiroshi committed:
Upgrade gopsutil for Apple silicon mac compatibility of go test (#342)

* Update gopsutil for Apple silicon mac compatibility of make test.

* go mod tidy

percona-csalguero committed:
PMM-8489 Fixed topology labels on mongos (#325)

* PMM-8489 Fixed topology labels on mongos

* Updated mog for go 1.16

* Updated dependency

* Updated toolkit dep to 3.x

* Updated go.sum for go 1.16

In my local I have go 1.17 so the sum was different and tests were
failing.

* Fixed mod conflicts

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

percona-csalguero committed:
PMM-8772 Enable DBStats (#341)

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

percona-csalguero committed:
PMM-7116 myState metric always has a value

percona-csalguero committed:
PMM-7116 Incorrect STARTUP state

percona-csalguero committed:
Merge branch 'main' into PMM-7116_incorrect_STARTUP_state

percona-csalguero committed:
Merge branch 'main' into PMM-7116_incorrect_STARTUP_state

percona-csalguero committed:
PMM-7116 Ran format

percona-csalguero committed:
Merge branch 'PMM-7116_incorrect_STARTUP_state' of percona.github.com:percona/mongodb_exporter into PMM-7116_incorrect_STARTUP_state

percona-csalguero committed:
PMM-8849 Fixed panic on nil data

When the data collection was slow for some reason like having a big
number of databases, one of the functions was returning a nil but that
condition was not being checked and a struct field was used causing a
panic.

percona-csalguero committed:
PMM-8849 Skiped test for GH Actions

percona-csalguero committed:
PMM-8849 updated test for GH Actions

percona-csalguero committed:
PMM-8849 Updated test for GH Actions

percona-csalguero committed:
PMM-8849 Updated test for GH Actions

percona-csalguero committed:
PMM-8849 Updated test for GH Actions

percona-csalguero committed:
PMM-8849 Updated test for GH Actions

percona-csalguero committed:
Removed irrelevant comment

denisok committed:
Add permissions (#344)

percona-csalguero committed:
Ran tidy

percona-csalguero committed:
Merge branch 'PMM-8539_cluster_role_label' into release-0.20.8

percona-csalguero committed:
Merge branch 'PMM-7116_incorrect_STARTUP_state' into release-0.20.8

percona-csalguero committed:
Merge branch 'PMM-8849_panic_on_timeout' into release-0.20.8

percona-csalguero committed:
Merge branch 'PMM-8849_panic_on_timeout' into release-0.20.8

percona-csalguero committed:
PMM-8874 Merged tickets for release

Merged:
- PMM-8849 Fixed panic on nil data
- PMM-8539_cluster_role_label
- PMM-8539 Fixed cluster role cluster
- PMM-7424 Add dbStatsCollector
- PMM-7116 Incorrect STARTUP state

Updated mongo driver and ci-lint

percona-csalguero committed:
Merge branch 'release-0.20.8' of percona.github.com:percona/mongodb_exporter into release-0.20.8

rfratto committed:
Remove dependency on github.com/prometheus/common/log (#355)

github.com/prometheus/common/log was removed in favor of Prometheus
internally using github.com/go-kit/kit/log (and later
github.com/go-kit/log). This exporter still uses the removed package,
which prevents downstream importers from being able to upgrade the
prometheus/common module.

Signed-off-by: Robert Fratto <robertfratto@gmail.com>

rfratto committed:
Expose exporter.Handler and allow disabling default registry (#356)

The Grafana Agent (https://github.com/grafana/agent) embeds
mongodb_exporter as an integration, where it needs to be able to call
into the exporter's HTTP handler for its all-in-one server. This
endpoint also needs to only return metrics for the exporter itself and
not from the global registry, which is populated by unrelated metrics.

Signed-off-by: Robert Fratto <robertfratto@gmail.com>

percona-csalguero committed:
PMM-7116 myState metric always has a value (#345)

* PMM-7116 myState metric always has a value

* PMM-7116 Incorrect STARTUP state

* PMM-7116 Ran format

* Removed irrelevant comment

percona-csalguero committed:
PMM-8539 cluster role label (#338)

* PMM-8539 Fixed cluster role cluster

Fixed following specs in https://jira.percona.com/browse/PMM-6494.
cl_role: Cluster role - "shardsvr", "configsvr", "mongos" or "".
Set "mongos" if there is a sharding.configDB config value set.
For others whatever the value for sharding.clusterRole is.
It will be empty when a non-sharded replicaset or standalone mongod node.

Added docker helpers.

denisok committed:
Revert "PMM-8539 cluster role label (#338)" (#359)

This reverts commit 6d02604b187508d70b5f7237dacfebd43ca3d5c5.

kbudde committed:
build container image on release (#354)

* build container image on release

ref #264 #64

Signed-off-by: Kris Budde <kris@budd.ee>

kbudde committed:
build container image on release (#354)

* build container image on release

ref #264 #64

Signed-off-by: Kris Budde <kris@budd.ee>

percona-csalguero committed:
Merge branch 'main' into release-0.20.8

percona-csalguero committed:
Updated go mod

percona-csalguero committed:
Updated format

percona-csalguero committed:
Updated format

percona-csalguero committed:
Updated format

percona-csalguero committed:
Release 0.20.8 (#353)

* PMM-8539 Fixed cluster role cluster

Fixed following specs in https://jira.percona.com/browse/PMM-6494.
cl_role: Cluster role - "shardsvr", "configsvr", "mongos" or "".
Set "mongos" if there is a sharding.configDB config value set.
For others whatever the value for sharding.clusterRole is.
It will be empty when a non-sharded replicaset or standalone mongod node.

* PMM-8539 Added docker helpers

* PMM-7116 myState metric always has a value

* PMM-7116 Incorrect STARTUP state

* PMM-7116 Ran format

* PMM-8849 Fixed panic on nil data

When the data collection was slow for some reason like having a big
number of databases, one of the functions was returning a nil but that
condition was not being checked and a struct field was used causing a
panic.

* PMM-8849 Skiped test for GH Actions

* PMM-8849 updated test for GH Actions

* Removed irrelevant comment

* Ran tidy

* PMM-8874 Merged tickets for release

Merged:
- PMM-8849 Fixed panic on nil data
- PMM-8539_cluster_role_label
- PMM-8539 Fixed cluster role cluster
- PMM-7424 Add dbStatsCollector
- PMM-7116 Incorrect STARTUP state

Updated mongo driver and ci-lint

* build container image on release (#354)

* build container image on release

ref #264 #64

Signed-off-by: Kris Budde <kris@budd.ee>


* Updated format

* Updated format

Co-authored-by: Kris Budde <kris.budde@gmail.com>

percona-csalguero committed:
PMM-8539 Updated documentation (#363)

* PMM-8539 Updated documentation

* PMM-8539 Clarification for topology labels

rnovikovP committed:
Update README.md (#368)

PMM-9024
Fix mongodb_exporter flags documentation

kbudde committed:
close client if initial connection failed (#367)

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

mas15 committed:
Issue-358 | Replace --enable.COLLECTOR with --collector.COLLECTOR (#365)

algestam committed:
README.md: Fix log level example (#370)

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>

kbudde committed:
prevent exporter exit after start (#366)

* prevent exporter exit after start

- with `--mongodb.global-conn-pool` option set, the exporter only tried to connect once after start
- topology is now refreshes for every scrape

* single use and global connection behavior is equal now

- globalConPool will try to connect on start as before
- this will create globalConn or log an error, but it will not stop the exporter
- globalConPool retries to connect on each scrape until success

* removed unneccessary new line

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>
Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

igroene committed:
PMM-8011 Add top admincommand collector (#357)

* PMM-8011 Add top admincommand collector

* format code properly

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

denisok committed:
fix version reported by community exporter, now empty (#373)

denisok committed:
fix version reported by community exporter, now empty (#373)

denisok committed:
Update issue templates and contribution guidelines (#374)

* Update issue templates and contribution guidelines

* adding coc, branching and adjusting templates

* fixing wording

* Update CONTRIBUTING.md

* Update .github/ISSUE_TEMPLATE/bug_report.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

* Update .github/ISSUE_TEMPLATE/bug_report.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

* Update .github/ISSUE_TEMPLATE/bug_report.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

* Update .github/ISSUE_TEMPLATE/feature_request.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

* Update .github/ISSUE_TEMPLATE/feature_request.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

* Update CODE_OF_CONDUCT.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

* Update .github/ISSUE_TEMPLATE/bug_report.md

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

Co-authored-by: Alex Tymchuk <alexander.tymchuk@percona.com>

Lp-Francois committed:
Add docker image link to the Readme (#351)

* Add docker image link to the Readme

As stated in this Issue: https://github.com/percona/mongodb_exporter/issues/64

A quick Docker documentation is a good enhancement to MongoDB exporter.

Related project card: https://github.com/orgs/percona/projects/2#card-66138726

* Update Readme according to the docker hub repo

* Remove useless comment

* Add docker instructions

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

algestam committed:
README.md: Improve notes on the previous versions of the exporter (#371)

Co-authored-by: Roma Novikov <roma.novikov@percona.com>
Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>

BupycHuk committed:
Changes from 2.23.0 (#372)

* PMM-8489 Fixed topology labels on mongos

* Updated mog for go 1.16

* Updated dependency

* Updated toolkit dep to 3.x

* Updated go.sum for go 1.16

In my local I have go 1.17 so the sum was different and tests were
failing.

* PMM-8539 Fixed cluster role cluster

Fixed following specs in https://jira.percona.com/browse/PMM-6494.
cl_role: Cluster role - "shardsvr", "configsvr", "mongos" or "".
Set "mongos" if there is a sharding.configDB config value set.
For others whatever the value for sharding.clusterRole is.
It will be empty when a non-sharded replicaset or standalone mongod node.

* PMM-8539 Added docker helpers

* Fixed mod conflicts

* PMM-7116 myState metric always has a value

* PMM-7116 Incorrect STARTUP state

* PMM-7116 Ran format

* PMM-8849 Fixed panic on nil data

When the data collection was slow for some reason like having a big
number of databases, one of the functions was returning a nil but that
condition was not being checked and a struct field was used causing a
panic.

* PMM-8849 Skiped test for GH Actions

* PMM-8849 updated test for GH Actions

* PMM-8849 Updated test for GH Actions

* PMM-8849 Updated test for GH Actions

* PMM-8849 Updated test for GH Actions

* PMM-8849 Updated test for GH Actions

* Removed irrelevant comment

* Ran tidy

* PMM-8874 Merged tickets for release

Merged:
- PMM-8849 Fixed panic on nil data
- PMM-8539_cluster_role_label
- PMM-8539 Fixed cluster role cluster
- PMM-7424 Add dbStatsCollector
- PMM-7116 Incorrect STARTUP state

Updated mongo driver and ci-lint

* Revert "Merge branch 'PMM-8489_topology_labels_on_mongos'"

This reverts commit 8905fce6897eb187ef20c3f0c51cc2d9fc4bf080, reversing
changes made to 45530849d127359b22ae803b4ece5a57a062f2e6.

* Run diff only on latest go version

* Updated go versions

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

Commit:
PMM-9168 remove the review workflow (#384)

* PMM-9168 remove the review workflow

* PMM-9168 updated links in PR template

dependabot[bot] committed:
Bump github.com/golangci/golangci-lint from 1.42.1 to 1.43.0 in /tools (#376)

Bumps [github.com/golangci/golangci-lint](https://github.com/golangci/golangci-lint) from 1.42.1 to 1.43.0.
- [Release notes](https://github.com/golangci/golangci-lint/releases)
- [Changelog](https://github.com/golangci/golangci-lint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/golangci/golangci-lint/compare/v1.42.1...v1.43.0)

---
updated-dependencies:
- dependency-name: github.com/golangci/golangci-lint
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

percona-csalguero committed:
Rename enable top (#387)

* PMM-8489 Fixed topology labels on mongos

* Updated mog for go 1.16

* Updated dependency

* Updated toolkit dep to 3.x

* Updated go.sum for go 1.16

In my local I have go 1.17 so the sum was different and tests were
failing.

* Fixed mod conflicts

* Revert "Merge branch 'PMM-8489_topology_labels_on_mongos'"

This reverts commit 8905fce6897eb187ef20c3f0c51cc2d9fc4bf080, reversing
changes made to 45530849d127359b22ae803b4ece5a57a062f2e6.

* Renamed enable.top to collector.topmetrics

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

BupycHuk committed:
Merge remote-tracking branch 'origin/pmm-2.24.0' into main

# Conflicts:
#	go.mod
#	tools/go.mod
#	tools/go.sum

percona-csalguero committed:
PMM-7802 arbiter authentication (#378)

* PMM-8489 Fixed topology labels on mongos

* Updated mog for go 1.16

* Updated dependency

* Updated toolkit dep to 3.x

* Updated go.sum for go 1.16

In my local I have go 1.17 so the sum was different and tests were
failing.

* Fixed mod conflicts

* Revert "Merge branch 'PMM-8489_topology_labels_on_mongos'"

This reverts commit 8905fce6897eb187ef20c3f0c51cc2d9fc4bf080, reversing
changes made to 45530849d127359b22ae803b4ece5a57a062f2e6.

* Arbiters don't have a Cluster ID and cannot run getDiagnosticData.
Anyway, other metrics can be retrieved so topology labels shouldn't
return an error if the cluster id cannot be determined, just return an
empty cluster id and don't returns errors for getDiagnosticData.

* PMM-7802 Updated test

* Updated test

* Removed blank line

* Updated tools

* Updated tools

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

percona-csalguero committed:
PMM-8770 enable safe metrics collection (#389)

* PMM-8489 Fixed topology labels on mongos

* Updated mog for go 1.16

* Updated dependency

* Updated toolkit dep to 3.x

* Updated go.sum for go 1.16

In my local I have go 1.17 so the sum was different and tests were
failing.

* Fixed mod conflicts

* Revert "Merge branch 'PMM-8489_topology_labels_on_mongos'"

This reverts commit 8905fce6897eb187ef20c3f0c51cc2d9fc4bf080, reversing
changes made to 45530849d127359b22ae803b4ece5a57a062f2e6.

* PMM-8770 Updated discovery mode

Now discovering mode automatically search for all databases and all
collections. It might receive a filter list and ignores system
collections.

* PMM-8770 Added common functions

* PMM-8770 New collstats limit

* Fixed race condition

* Top collector won't be enabled on mongos

* Fixes for CR

* Renamed collector.topmetric param

Co-authored-by: JiriCtvrtka <62988319+JiriCtvrtka@users.noreply.github.com>

anton-bystrov committed:
Add build deb and rpm packages. (#392)

* Add build deb and rpm packages.

* Remove autostart, add example config file.

* Change create user

* Remove rename package for mac

dependabot[bot] committed:
Bump golang.org/x/tools from 0.1.7 to 0.1.8 in /tools (#390)

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.1.7 to 0.1.8.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.1.7...v0.1.8)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>

percona-csalguero committed:
PMM-9312 default collectors (#396)

* release-0.30.0

* PMM-9312 Added new filters for discovering mode

* Added flags for collstats & indexstats

* Update main.go

Co-authored-by: Nurlan Moldomurov <nurlan.moldomurov@percona.com>

* Changes for CR

* Updates for CR

* Ran format

* Fixed top collector flag

* Updated filters

* PMM-9312 Fix MongoFilter (#399)

Co-authored-by: Nurlan Moldomurov <nurlan.moldomurov@percona.com>

percona-csalguero committed:
trigger

percona-csalguero committed:
Using this branch for release

rlankfo committed:
serve metrics handler with prometheus exporter-toolkit (#393)

* serve metrics handler with prometheus exporter-toolkit

* remove go-kit/log

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

percona-csalguero committed:
PMM-9400 Fix filters for PMM-9312 (#402)

* PMM-9400 Fix filters for PMM-9312

* Added a comment

* Improvement for CR

* Updates for CR

* Fixes for CR

* Changes for CR

* Changes for CR

percona-csalguero committed:
PMM-9400 Fix filters for PMM-9312 (#402) (#408)

* PMM-9400 Fix filters for PMM-9312

* Added a comment

* Improvement for CR

* Updates for CR

* Fixes for CR

* Changes for CR

* Changes for CR

percona-csalguero committed:
Revert "serve metrics handler with prometheus exporter-toolkit (#393)" (#409)

This reverts commit cb94884dd53888238d44933f341cbcd89ebdd134.

percona-csalguero committed:
Revert "Revert "serve metrics handler with prometheus exporter-toolkit (#393)" (#409)" (#410)

PR-393 was merged before we can use it properly from pmm-managed.
In order to make it work with basic auth, we need to implement the
changes in pmm-managed/mongodb exporter config to create a new file
having basic auth credentials.

This reverts commit 5e5bf25c7cfe3946ef6935632ab3d649db2ecc85.

dependabot[bot] committed:
Bump go.mongodb.org/mongo-driver from 1.7.2 to 1.8.2 (#405)

Bumps [go.mongodb.org/mongo-driver](https://github.com/mongodb/mongo-go-driver) from 1.7.2 to 1.8.2.
- [Release notes](https://github.com/mongodb/mongo-go-driver/releases)
- [Commits](https://github.com/mongodb/mongo-go-driver/compare/v1.7.2...v1.8.2)

---
updated-dependencies:
- dependency-name: go.mongodb.org/mongo-driver
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

dependabot[bot] committed:
Bump github.com/reviewdog/reviewdog from 0.13.0 to 0.13.1 in /tools (#401)

Bumps [github.com/reviewdog/reviewdog](https://github.com/reviewdog/reviewdog) from 0.13.0 to 0.13.1.
- [Release notes](https://github.com/reviewdog/reviewdog/releases)
- [Changelog](https://github.com/reviewdog/reviewdog/blob/master/CHANGELOG.md)
- [Commits](https://github.com/reviewdog/reviewdog/compare/v0.13.0...v0.13.1)

---
updated-dependencies:
- dependency-name: github.com/reviewdog/reviewdog
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/alecthomas/kong from 0.2.17 to 0.3.0 (#404)

Bumps [github.com/alecthomas/kong](https://github.com/alecthomas/kong) from 0.2.17 to 0.3.0.
- [Release notes](https://github.com/alecthomas/kong/releases)
- [Commits](https://github.com/alecthomas/kong/compare/v0.2.17...v0.3.0)

---
updated-dependencies:
- dependency-name: github.com/alecthomas/kong
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>

idoqo committed:
PMM-5471 add license checks (#413)

* PMM-5471 fix files without license headers

* PMM-5471 include license check in github workflow

* PMM-5471 update copyright year

* PMM-5471 support multiple copyright years in license checker

* PMM-5471 ignore copyright year in check

* PMM-5471 use regexp to check copyright pattern

* PMM-5471 restrict regex year

BupycHuk committed:
Merge branch 'main' into pmm-2.26.0

BupycHuk committed:
Update common.go

JiriCtvrtka committed:
PMM-9471 Context deadline (#420)

* PMM-9471 Cover case with limit collections -1 (auto).

* PMM-9471 Logic, nil fixies. Reflect timeout from VM.

* PMM-9471 Revert dynamic timeout for now.

* Revert "PMM-9471 Revert dynamic timeout for now."

This reverts commit 02cb9b721cae75c56134602b2c8638c43520676e.

* PMM-9471 New default timeout.

* PMM-9471 Fix missing flag.

* PMM-9471 support of collect[] query parameter

* PMM-9471 Another changes in exporter. Context background.

Co-authored-by: Nurlan Moldomurov <bupychuk1989@gmail.com>

BupycHuk committed:
Update common.go

Commit:
Revert "Revert "Revert "serve metrics handler with prometheus exporter-toolkit (#393)" (#409)" (#410)"

This reverts commit 239ecfd5f1e76d1a8cd7d2581fdb5e09b03e9449.

JiriCtvrtka committed:
PMM-9507 Better help. (#428)

* PMM-9507 Better help.

* PMM-9507 Fix.

JiriCtvrtka committed:
PMM-9507 Better help. (#429)

JiriCtvrtka committed:
PMM-9458 Disable DBStats and TopMetrics where limit is active. (#417)

* PMM-9458 Proposed solution

* PMM-9458 Improve help-

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

JiriCtvrtka committed:
PMM-9458-9471 Disable collectors on limit and context deadline (#431)

* PMM-9458 Disable DBStats and TopMetrics where limit is active. (#417)

* PMM-9458 Proposed solution

* PMM-9458 Improve help-

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>

* PMM-9471 Context deadline (#420)

* PMM-9471 Cover case with limit collections -1 (auto).

* PMM-9471 Logic, nil fixies. Reflect timeout from VM.

* PMM-9471 Revert dynamic timeout for now.

* Revert "PMM-9471 Revert dynamic timeout for now."

This reverts commit 02cb9b721cae75c56134602b2c8638c43520676e.

* PMM-9471 New default timeout.

* PMM-9471 Fix missing flag.

* PMM-9471 support of collect[] query parameter

* PMM-9471 Another changes in exporter. Context background.

Co-authored-by: Nurlan Moldomurov <bupychuk1989@gmail.com>

* PMM-9458 Proper merge.

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: Nurlan Moldomurov <bupychuk1989@gmail.com>

Commit:
collstats: fix compatibility issue and disable latency histograms (#433)

$collStats is invoked by exporter to get collection statistics from
MongoDB. The configuration generated by exporter worked for v4.x
of MongoDB, but failed for v5.0. This change modifies configuration
 to work for v4.x and v5.0.

Once fix works, latency statistics are generated in form of histogram.
MongoDB exporter doesn't generate histogram metrics for Prometheus
as of yet. Hence we disable histogram for latency statistics in this patch.

dependabot[bot] committed:
Bump go.mongodb.org/mongo-driver from 1.8.2 to 1.8.3 (#434)

Bumps go.mongodb.org/mongo-driver from 1.8.2 to 1.8.3.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

dependabot[bot] committed:
Bump golang.org/x/tools from 0.1.8 to 0.1.9 in /tools (#427)

Bumps golang.org/x/tools from 0.1.8 to 0.1.9.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

dependabot[bot] committed:
Bump github.com/alecthomas/kong from 0.3.0 to 0.4.0 (#425)

Bumps github.com/alecthomas/kong from 0.3.0 to 0.4.0.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.11.0 to 1.12.1 (#424)

Bumps github.com/prometheus/client_golang from 1.11.0 to 1.12.1.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

Commit:
Update test matrix for CI pipeline (#441)

Test matrix should test exporter against all versions of
MongoDB which are officially supported. Hence MongoDB 5.0
is added. MongoDB 3.6 versions are removed, since they
have hit EOL.

Commit:
tools: Update build process (#440)

* Use go generate for tools build
* Remove usage of gofumports

Commit:
Fix CI pipeline warnings (#446)

- gofumpt: Remove deprecated flag
- golangci-lint: Disable deprecated linters; add reason in config

dependabot[bot] committed:
Bump mvdan.cc/gofumpt from 0.2.1 to 0.3.0 in /tools (#445)

Bumps [mvdan.cc/gofumpt](https://github.com/mvdan/gofumpt) from 0.2.1 to 0.3.0.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

aravandi committed:
[Dev-infra]Fix a minor bug in dev-infra setup (#437)

Fix a minor bug on the dev-infra setup. In MongoDB, config servers are initiated with --configsvr flag. The mongo-cnf-setup container has to use this flag as well.

dependabot[bot] committed:
Bump go.mongodb.org/mongo-driver from 1.8.3 to 1.8.4 (#449)

Bumps go.mongodb.org/mongo-driver from 1.8.3 to 1.8.4.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

BupycHuk committed:
Merge remote-tracking branch 'origin/main' into pmm-2.26.0

# Conflicts:
#	exporter/exporter.go
#	go.mod
#	go.sum

BupycHuk committed:
Merge conflicts fixed

BupycHuk committed:
Merge pull request #421 from percona/pmm-2.26.0

Pmm 2.26.0

dependabot[bot] committed:
Bump github.com/stretchr/testify from 1.7.0 to 1.7.1 (#453)

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.7.0 to 1.7.1.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.7.0...v1.7.1)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
PMM-9459 Remove usage of DescribeByCollect (#443)

DescribeByCollect method internally uses Collect method. This causes
exporter to retrieve metric data from database twice per scrape. This
patch reduces monitoring load on database server, as well as reduces
CPU and memory usage of exporter.

Changes to Describe and Collect methods has introduced a dependency.
Before a call to Collect method, Describe method  needs to be called.
Tests need to be updated to handle this assumption.

percona-csalguero committed:
PMM-9652 Fixed anti-pattern metric names (#450)

* PMM-9652 WIP initial test

* PMM-9652 Updated top exporter

Now metric names are properly built and the namespace is in a label.

* PMM-9652 Fixed anti-pattern metric names

Metric names shouldn't vahe variable data like database or collection
name as part of the name. The namespace and other information like the
index nane should be set in a label.
This patch fixes the metric names in collstas, indexstats and topstats
collectors.

* Ran format

* PMM-9652 Updated labels to have database+collection

* Updated labels for top collector

* Removed old comment

* Format

aravandi committed:
[feature] Add a Descending index metric name override flag (#435)

Fields with descending order in a MongoDB index have '-1' in index name.
'-' is not a valid character in data model for Prometheus. Hence it will be
replaced with 'DESC' to remove ambiguity between indexes with use of
new flag in exporter.

Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

Commit:
Updated CHANGELOG (#464)

percona-csalguero committed:
PMM-9757 Fixed panic on lost connection (#460)

Changed the way we instantiate the topology info object.
Now we call the newTopologyInfo and it never returns an error and the
object is never nill. If it is not possible to connect, topology labels
remain empty but we return a valid and not nill object.
Also added context termination check to avoid the ugly stack traces
produced by context timeouts.

Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>

Commit:
PMM-9757 Fixed panic on lost connection (#465)

Changed the way we instantiate the topology info object.
Now we call the newTopologyInfo and it never returns an error and the
object is never nill. If it is not possible to connect, topology labels
remain empty but we return a valid and not nill object.
Also added context termination check to avoid the ugly stack traces
produced by context timeouts.

Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: Denys Kondratenko <denisok@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/tools from 0.1.9 to 0.1.10 in /tools (#454)

Bumps golang.org/x/tools from 0.1.9 to 0.1.10.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

dependabot[bot] committed:
Bump mvdan.cc/gofumpt from 0.3.0 to 0.3.1 in /tools (#463)

Bumps mvdan.cc/gofumpt from 0.3.0 to 0.3.1.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/alecthomas/kong from 0.4.0 to 0.5.0 (#451)

Bumps github.com/alecthomas/kong from 0.4.0 to 0.5.0.

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Carlos Salguero <carlos.salguero@percona.com>
Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

dependabot[bot] committed:
Bump percona-platform/checkout from 2 to 3 (#466)

Bumps percona-platform/checkout from 2 to 3.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

BupycHuk committed:
Update postinst (#468)

Co-authored-by: Shashank Sinha <shashank.sinha@percona.com>

BupycHuk committed:
Merge pull request #473 from percona/pmm-2.27.0

Changes from 2.27.0

arggh committed:
Update readme regarding `enable.top` (#469)

'enable.top' was renamed to 'collector.topmetrics'. This updates the README to reflect that change.

dependabot[bot] committed:
Bump github.com/golangci/golangci-lint from 1.44.2 to 1.45.2 in /tools (#462)

Bumps github.com/golangci/golangci-lint from 1.44.2 to 1.45.2.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

percona-csalguero committed:
PMM-9450 Enable basic auth (#411)

PMM-9450 Added webconfig auth implementation

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.32.1 to 0.34.0 (#478)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.32.1 to 0.34.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.32.1...v0.34.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

csibar committed:
Exclude awaitable isMaster commands

So they don't get reported as long running queries
for mongodb_operations_longest_running_query_seconds.

See https://jira.mongodb.org/browse/SERVER-51358
